redisgears_core/src/lib.rs

```diff
@@ -673,27 +742,21 @@ fn js_init(ctx: &Context, args: &[RedisString]) -> Status {
 
 const fn js_info(_ctx: &InfoContext, _for_crash_report: bool) {}
 
-pub(crate) fn verify_oom(flags: u8) -> bool {
-    if (flags & FUNCTION_FLAG_NO_WRITES) == 0 {
+pub(crate) fn verify_oom(flags: FunctionFlags) -> bool {
+    if !flags.contains(FunctionFlags::NO_WRITES) {
```

Comment:
So this function need to return true if we are OK (did not reach OOM) and false if we are OOM.
If NO_WRITE flag is on, we are for sure OK because we can not bypass memory limit if we are not doing any writes.
If NO_WRITE is off, but the function state that it is OK to bypass the memory limit (ALLOW_OOM is on) then it is also ok.
If both are are off then we should check that we did not reach OOM.

So I believe you are right and we can change it.